cmd/goal/application.go

```diff
@@ -1172,17 +1178,47 @@ var methodAppCmd = &cobra.Command{
 		// Parse transaction parameters
 		appArgsParsed, appAccounts, foreignApps, foreignAssets := getAppInputs()
 		if len(appArgsParsed) > 0 {
-			reportErrorf("in goal app method: --arg and --app-arg are mutually exclusive, do not use --app-arg")
+			reportErrorf("--arg and --app-arg are mutually exclusive, do not use --app-arg")
+		}
+
+		// Construct schemas from args
+		localSchema := basics.StateSchema{
+			NumUint:      localSchemaUints,
+			NumByteSlice: localSchemaByteSlices,
+		}
+
+		globalSchema := basics.StateSchema{
+			NumUint:      globalSchemaUints,
+			NumByteSlice: globalSchemaByteSlices,
 		}
 
 		onCompletionEnum := mustParseOnCompletion(onCompletion)
```

Comment:
This is obviously a decision that was made before, this PR doesn't change it.  But do you think we should really be requiring on-complete to be set always?  Rather than default to no-op?

Reply:
Actually you don't need to explicitly pass it in, it will default to "NoOp": https://github.com/algorand/go-algorand/blob/1642a3214a12cd49843d62b2e18a1d3e516e7316/cmd/goal/application.go#L125

The e2e test for creation does not pass in `--on-completion`.

Reply:
oh, yeah. I guess I wish there was no "must" on that method name then.  While it's true, it seems like it's trying to convey something important that never comes up in practice because of the default. But I don't care _that_ much.

Reply:
I agree it's a bit misleading, but I'll leave it for now since there are multiple other helper methods in the goal code that use "must" to convey they will either succeed or panic.